pennylane/gradients/vjp.py

```diff
@@ -46,6 +46,7 @@ def compute_vjp(dy, jac, num=None):
 
     if not isinstance(dy_row, np.ndarray):
         jac = math.convert_like(jac, dy_row)
+        jac = math.cast(jac, dy_row.dtype)

```

Comment:
very minor @antalszava, but I'm curious why `cast_like` wasn't used here, I thought it would be safer that using `dy_row.dtype`?

Reply:
`qml.math.cast_like` assumes that the two tensors are already of the same type if I'm not mistaken. I ran into issues with it using Torch.